openvdb_houdini/houdini/SOP_OpenVDB_Advect_Points.cc

```diff
@@ -1107,25 +1111,18 @@ VDB_NODE_OR_CACHE(VDB_COMPILABLE_SOP, SOP_OpenVDB_Advect_Points)::evalAdvectionP
     UT_String ptGroupStr;
     evalString(ptGroupStr, "group", 0, now);
 
-    if (advectVdbPoints)
-    {
-        parms.mPointGroup = 0;
-        parms.mGroup = matchGroup(*parms.mPointGeo, ptGroupStr);
+    parms.mPointGroup = parsePointGroups(ptGroupStr, GroupCreator(gdp));
+    parms.mVDBGroup = matchGroup(*parms.mPointGeo, evalStdString("vdbgroup", now));
```

Comment:
I think we still need to guard mVDBGroup with an advectVdbPoints test.  We are not using this group if the advectVdbPoints is false, but matchGroups will throw an exception that will red flag the node if it is invalid.  So there is not just the trivial cost of computing the group; but the potential a node may flag red due to an invisible parameter value.  (As an aside, if these were using the .ds-style verbs disabled parameters would always evaluate to defaults so this would be safe; but I believe the current VDB shim will fully evaluate the parameters regardless of the disabled state)

Reply:
Yes, good point, you're right, I can fix that. 

I'm still a bit wary about using the ds stuff as I've worked with other autogen c++ tools in the past and always found them quite fiddly to work with, but perhaps I just need to stick with it. I'm sure proper Make/CMake support for this in VDB would help. Certainly something worth discussing in our TSC meetings at some point though if you think there's a significant enough benefit to make it worthwhile making the transition or if it simplifies the integration of VDB into Houdini.

Reply:
I certainly agree with your suspicious on autogen c++ tools.

The first benefit of ds-style is that it exactly matches how HDAs store parameters.  This ensures there is always compatibility between what can be exported and what can be created.  The second is that it uses the disablewhen syntax.  This allows the parameter evaluation to be restricted to enabled parameters, avoiding any false dependencies and returning defaults.

But since VDB nodes themselves tend to be heavyweight, there isn't much benefit in converting them all to .ds.  It would be a big sweep.

Reply:
Good to know, thanks.

Reply:
@jmlait @danrbailey just to mention here that the biggest benefit of converting to .ds is to resolve disabled time dependant parameters. Whilst this luckily doesn't seem to be common on many VDB nodes it still is significant and would be good to resolve, otherwise you end up cooking every frame!